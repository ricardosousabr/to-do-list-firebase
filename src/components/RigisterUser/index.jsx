import { getAuth, createUserWithEmailAndPassword } from "firebase/auth";
import { initializeApp } from 'firebase/app'
import firebaseConfig from '../../../firebase/config.js'
import { useState } from "react";

export default function RigisterUser() {
  const app = initializeApp(firebaseConfig)
  const [emalUser, setEmailUser] = useState("")
  const [passwordUser, setPasswordUser] = useState("")

  function addUser() {
    const auth = getAuth();
    createUserWithEmailAndPassword(auth, emalUser, passwordUser)
      .then(() => {
        alert("seja bem vindo")
      })
      .catch((error) => {
        console.log(error)
      });
  }

  function handleSubmit(e) {
    e.preventDefault()
    addUser()
  }

  return (
    <>
      <div>
        <h1>Se registre</h1>
      </div>
      <div>
        <form onClick={handleSubmit}>
          <div>
            <input placeholder="e-mal" type="text" onChange={(e) => { setEmailUser(e.target.value) }} />
          </div>
          <div>
            <input placeholder="password" type="text" onChange={(e) => { setPasswordUser(e.target.value) }} />
          </div>
          <div>
            <button>Registrar</button>
          </div>
        </form>
      </div>
    </>
  )
}
