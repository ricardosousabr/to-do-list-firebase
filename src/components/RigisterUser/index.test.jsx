/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import RigisterUser from './index'

it('render registration form', () => {
  const { container } = render(<RigisterUser />)
  expect(container).toMatchSnapshot()
})
