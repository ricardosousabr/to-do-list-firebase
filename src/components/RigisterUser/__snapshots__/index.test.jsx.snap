// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render registration form 1`] = `
<div>
  <div>
    <h1>
      Se registre
    </h1>
  </div>
  <div>
    <form>
      <div>
        <input
          placeholder="e-mal"
          type="text"
        />
      </div>
      <div>
        <input
          placeholder="password"
          type="text"
        />
      </div>
      <div>
        <button>
          Registrar
        </button>
      </div>
    </form>
  </div>
</div>
`;
